src/ai/backend/storage/api/vfolder/types.py

```diff
@@ -0,0 +1,130 @@
+import uuid
+from pathlib import Path, PurePath, PurePosixPath
+from typing import List, Optional, TypeAlias
+
+from pydantic import AliasChoices, Field
+from pydantic import BaseModel as PydanticBaseModel
+
+from ai.backend.common.types import BinarySize, QuotaConfig, QuotaScopeID, VFolderID
+
+
+class BaseModel(PydanticBaseModel):
+    """Base model for all models in this module"""
+
+    model_config = {"arbitrary_types_allowed": True}
```

Comment:
```suggestion
    model_config = ConfigDict(arbitrary_types_allowed = True)
```
1. How about using `pydantic.ConfigDict`?
2. Why is the config needed for all models in this module?
3. Let's rename this class to `_BaseModel` to represent that it is private if it is used only in this `types` module